lib/core/connection/pool.js

```diff
@@ -25,12 +25,14 @@ const makeStateMachine = require('../utils').makeStateMachine;
 const DISCONNECTED = 'disconnected';
 const CONNECTING = 'connecting';
 const CONNECTED = 'connected';
+const DRAINING = 'draining';
 const DESTROYING = 'destroying';
 const DESTROYED = 'destroyed';
 const stateTransition = makeStateMachine({
-  [DISCONNECTED]: [CONNECTING, DESTROYING, DISCONNECTED],
-  [CONNECTING]: [CONNECTING, DESTROYING, CONNECTED, DISCONNECTED],
-  [CONNECTED]: [CONNECTED, DISCONNECTED, DESTROYING],
+  [DISCONNECTED]: [CONNECTING, DRAINING, DISCONNECTED],
+  [CONNECTING]: [CONNECTING, CONNECTED, DRAINING, DISCONNECTED],
+  [CONNECTED]: [CONNECTED, DISCONNECTED, DRAINING],
+  [DRAINING]: [DRAINING, DESTROYING, DESTROYED],
```

Comment:
What does Draining do?

Reply:
`DRAINING` is distinguished from `DESTROYING` in that no new commands can be written to the pool (it gates `Pool.prototype.write`), but new connections can be made. This is so that we can accommodate fire-and-forget commands, like our own call to [endSessions](https://github.com/mongodb/node-mongodb-native/blob/5025159f4a7fe3ea673f2570ffa60ae36361095c/lib/core/sessions.js#L597) during a `Topology.close`. The command is queued for dispatch, but the pool might be at 0 connections (perhaps due to a previous error), so the pool needs to spin up a connection in graceful shutdown mode.